src/apps/ENGINE/src/compiler.cpp

```diff
@@ -1347,27 +1339,22 @@ bool COMPILER::Compile(SEGMENT_DESC &Segment, char *pInternalCode, uint32_t pInt
             DebugSourceLine = DSL_INI_VALUE;
             strcpy_s(DebugSourceFileName, Token.GetData());
             break;
-        case INCLUDE_LIBRIARY:
+        case INCLUDE_LIBRIARY: {
             SCRIPT_LIBRIARY *pLib;
-            SLIBHOLDER *pH;
-            bool bFound;
             VMA *pClass;
 
             //-----------------------------------------------------
             // check if already loaded
-            bFound = false;
-            for (n = 0; n < LibriaryFuncs.GetClassesNum(); n++)
+            auto name = std::string_view(Token.GetData());
+            auto comparator = [name](const auto &library)
+            {
+                return storm::iEquals(library.name, name);
```

Comment:
This is fine now but still a bit weird. A simple range-based with boolean variable would look more descriptive...
But if you want to avoid it anyway in favor of functional style, there is a pattern using upper_bound with the desired binary predicate:
`if(std::ranges::upper_bound(LibriaryFuncs, name, storm, iEquals) != std::end(LibriaryFuncs)) {
  break;
}`
https://godbolt.org/z/c789hP8EG
Consider using it instead. 
Everything else is just fine 👍